### DISCUSSION TITLE
Why there is no activation on the last layer of ResNET block?

### DISCUSSION THREAD

**Discussion by IIIBlueberry** [junyanz/pytorch-CycleGAN-and-pix2pix#1355] [General]
[Current implementation of this git ResNET block](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/models/networks.py#L376) is [x>Conv>Norm>ReLU>Conv>Norm>Add x] shouldn't there should be an activation layer after 'Add x' so that the next resnet block first Conv input is Activated(**relu(x)**)

Or this resnet block implementation supposed to have pre-activation layer?
![image](https://miro.medium.com/max/2000/1*kxgT5JHFEB31auMdj1iwaA.png)

**Comment by junyanz** [comment]
We follow the ResNet block design used by Johnson et al's amazing repo https://github.com/jcjohnson/fast-neural-style, which contains two Conv layers per block. The word "resnet block" might be overladed in different papers. We refer you to the original fast style transfer [paper](https://cs.stanford.edu/people/jcjohns/eccv16/) for more details.
